### DISCUSSION TITLE
What is the correct way to use a boolean modifier in Blender for a nif that will be exported with Nifly?

### DISCUSSION THREAD

**Discussion by SonicMagna** [BadDogSkyrim/PyNifly#156] [Q&A]
(In Blender 3.5) I have cube A and cube B. Cube B is smaller than A and we use it as a boolean modifier (Difference option) to make A into an 'open box'. After clicking Apply for the boolean and delete cube B, cube A is an open box. Everything looks fine. When I select cube A and export, the resulting nif is cube A plus the overlapping part of Cube B. It's still a solid cube, not the open box I see in Blender. In the nif I can see the edges of cube B inside cube A.
I have tried with the 'export modifier' in nifly on and off, but when I click Apply for the boolean, the modifier disappears so attempting to export with 'export modifiers' generates an error since there aren't any modifiers any more.
I know I'm missing something obvious.
TIA

**Comment by BadDogSkyrim** [comment]
I'll have to try this, but 'export modifer' may not be able to handle this situation. Make a copy, apply the modifier, and export that shape.

'Export modifiers' shouldn't really generate an error if you don't have modifiers.
